stubs/docutils/docutils/writers/__init__.pyi

```diff
@@ -1,3 +1,28 @@
-from _typeshed import Incomplete
+from typing import Any, Generic, TypeVar
 
-def __getattr__(name: str) -> Incomplete: ...
+from docutils import Component, nodes
+from docutils.io import Output
+from docutils.languages import LanguageImporter
+
+_S = TypeVar("_S")
+
+class Writer(Component, Generic[_S]):
+
+    document: nodes.document | None = None
+
+    output: _S | None = None
+
+    language: LanguageImporter | None = None
+
+    destination: Output | None = None
+
+    parts: dict[str, Any]
+
+    def __init__(self) -> None: ...
+    def write(self, document: nodes.document, destination: Output) -> str | bytes | None: ...
+    def translate(self) -> None: ...
+    def assemble_parts(self) -> None: ...
```

Comment:
The symmetric-ish `docutils.Reader` seems to be type-documented in a way that presents attributes in approximate logical usage-order (configuration, initialization, prep, action(s)); if we want to match that, I'd suggest:

```suggestion
class Writer(Component, Generic[_S]):
    parts: dict[str, Any]
    language: LanguageImporter | None = None
    def __init__(self) -> None: ...
    document: nodes.document | None = None
    destination: Output | None = None
    output: _S | None = None
    def assemble_parts(self) -> None: ...a
    def translate(self) -> None: ...
    def write(self, document: nodes.document, destination: Output) -> str | bytes | None: ...
```